src/components/tl/mlx5/mcast/tl_mlx5_mcast_rcache.c

```diff
@@ -0,0 +1,161 @@
+/**
+ * Copyright (c) 2022-2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ *
+ * See file LICENSE for terms.
+ */
+
+#include "tl_mlx5_mcast_rcache.h"
+
+static ucc_status_t ucc_tl_mlx5_mcast_coll_reg_mr(ucc_tl_mlx5_mcast_coll_context_t *ctx,
+                                                  void *data, size_t data_size, void **mr)
```

Comment:
`mr` should be typed `struct ibv_mr**`

Reply:
I would rather keep the code as is. Because:
 
1) Imagine in the future, we support different IB memory registration mechanisms and each provides different kinds of memory region structs and we would like to have a single function declaration for both of them.

2) Since ibv_reg_mr is defined in a third-party library, at any moment, it is possible that they change the struct names of the outputs. We do not want to keep changing the function declarations inside our .h file as well as its implementation each time a third-party library changes its struct names.

Reply:
ok!